cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -801,4 +805,27 @@ private Set<Class<?>> addExtraDependencies(CConfiguration cConf, Set<Class<?>> d
 
     return dependencies;
   }
+
+  @VisibleForTesting
+  static void copyRuntimeToken(LocationFactory locationFactory, Arguments args, File tempDir,
+                               Map<String, LocalizeResource> localizeResources) throws IOException {
+    if (args.hasOption(ProgramOptionConstants.PEER_NAME)) {
```

Comment:
I think it's a little cleaner to have the caller do this check, otherwise the name makes it sound like it always copies.

FYI, for situations like these, it's also better for readability to invert the check:

```
if (!args.hasOption(...)) {
  return;
}
// logic here
```